cmd/handlers/update/handler.go

```diff
@@ -0,0 +1,91 @@
+package update
+
+import (
+	"github.com/kholodmv/go-service/cmd/common"
+	"github.com/kholodmv/go-service/cmd/storage"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type Handler struct {
+	repository storage.MetricRepository
+}
+
+func NewHandler(repository storage.MetricRepository) *Handler {
+	return &Handler{
+		repository: repository,
+	}
+}
+
+func (mh *Handler) UpdateMetric(res http.ResponseWriter, req *http.Request) {
+	common.CheckPostHTTPMethod(res, req)
+	res.Header().Set("Content-Type", "text/plain")
+
+	params := parsParams(res, req)
+
+	metricType := params[2]
+	checkType(res, metricType)
+
+	metricName := params[3]
+	checkName(res, metricName)
+
+	metricValue := params[4]
+	checkMetricsValue(res, metricValue, metricType)
+
+	saveMetrics(metricType, metricName, metricValue, mh)
+
+	res.WriteHeader(http.StatusOK)
+}
+
+func parsParams(res http.ResponseWriter, req *http.Request) []string {
+	parts := strings.Split(req.URL.Path, "/")
+	if len(parts) != 5 {
+		http.Error(res, "Invalid request", http.StatusNotFound)
+		return nil
+	}
+	return parts
+}
+
+func checkType(res http.ResponseWriter, metricType string) {
+	if metricType != common.Gauge && metricType != common.Counter {
```

Comment:
Не ошибка, но удобней для чтения 
```suggestion
	if metricType != common.Gauge &&
	 metricType != common.Counter {
```